File: Hard/2026-04-03-3661-Maximum-Walls-Destroyed-by-Robots/solution.py
class Solution:
    def maximumRobots(self, robots: List[int], distance: List[int], walls: List[int]) -> int:
        wall_set = set(walls)
        max_walls_destroyed = 0
        
        # Create a list of robot positions with their effective ranges
        robot_ranges = sorted((robots[i], robots[i] - distance[i], robots[i] + distance[i]) for i in range(len(robots)))
        
        # Set to track destroyed walls
        destroyed_walls = set()
        
        for pos, left_limit, right_limit in robot_ranges:
            # Check left direction
            for wall in range(left_limit, pos + 1):
                if wall in wall_set:
                    destroyed_walls.add(wall)
            # Check right direction
            for wall in range(pos, right_limit + 1):
                if wall in wall_set:
                    destroyed_walls.add(wall)
        
        return len(destroyed_walls)

File: Hard/2026-04-03-3661-Maximum-Walls-Destroyed-by-Robots/solution.js
var maximumWallsDestroyed = function(robots, distance, walls) {
    const wallSet = new Set(walls);
    const destroyedWalls = new Set();

    const robotData = robots.map((pos, idx) => ({
        pos,
        leftLimit: pos - distance[idx],
        rightLimit: pos + distance[idx]
    }));

    robotData.sort((a, b) => a.pos - b.pos);

    for (const { pos, leftLimit, rightLimit } of robotData) {
        // Check left direction
        for (let wall = leftLimit; wall <= pos; wall++) {
            if (wallSet.has(wall)) {
                destroyedWalls.add(wall);
            }
        }
        
        // Check right direction
        for (let wall = pos + 1; wall <= rightLimit; wall++) {
            if (wallSet.has(wall)) {
                destroyedWalls.add(wall);
            }
        }
    }

    return destroyedWalls.size;
};